File: src/main/java/openmedstack/readmodels/IUpdateReadModel.java
package openmedstack.readmodels;

import openmedstack.MessageHeaders;
import openmedstack.events.BaseEvent;

import java.util.concurrent.CompletableFuture;

public interface IUpdateReadModel<T extends BaseEvent> {
    CompletableFuture update(T domainEvent, MessageHeaders headers);
}

File: modules/openmedstack/src/main/kotlin/org/openmedstack/readmodels/IUpdateReadModel.kt
package org.openmedstack.readmodels

import org.openmedstack.MessageHeaders
import org.openmedstack.events.BaseEvent
import java.util.concurrent.CompletableFuture

interface IUpdateReadModel<T> where T: BaseEvent {
    fun canUpdate(eventType: Class<*>): Boolean
    fun <T> update(domainEvent: T, headers: MessageHeaders?): CompletableFuture<*>
}